sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/ChatThreadsImpl.java

```diff
@@ -1959,6 +1868,151 @@ public ChatThreadProperties getChatThreadProperties(String chatThreadId) {
+            String chatThreadId, TypingNotificationOptions sendTypingNotificationRequest, Context context) {
+        final String accept = "application/json";
+        return service.sendTypingNotification(
+                this.client.getEndpoint(),
+                chatThreadId,
+                this.client.getApiVersion(),
+                sendTypingNotificationRequest,
+                accept,
+                context);
+    }
+
+    /**
+     * Posts a typing event to a thread, on behalf of a user.
+     *
+     * @param chatThreadId Id of the thread.
+     * @param sendTypingNotificationRequest Details of the typing notification request.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> sendTypingNotificationAsync(
+            String chatThreadId, TypingNotificationOptions sendTypingNotificationRequest) {
+        return sendTypingNotificationWithResponseAsync(chatThreadId, sendTypingNotificationRequest)
+                .flatMap((Response<Void> res) -> Mono.empty());
+    }
+
+    /**
+     * Posts a typing event to a thread, on behalf of a user.
+     *
+     * @param chatThreadId Id of the thread.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> sendTypingNotificationAsync(String chatThreadId) {
+        final TypingNotificationOptions sendTypingNotificationRequest = null;
+        return sendTypingNotificationWithResponseAsync(chatThreadId, sendTypingNotificationRequest)
+                .flatMap((Response<Void> res) -> Mono.empty());
+    }
+
+    /**
+     * Posts a typing event to a thread, on behalf of a user.
+     *
+     * @param chatThreadId Id of the thread.
+     * @param sendTypingNotificationRequest Details of the typing notification request.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> sendTypingNotificationAsync(
+            String chatThreadId, TypingNotificationOptions sendTypingNotificationRequest, Context context) {
```

Comment:
should this be named options instead of sendTypingNotificationRequest ?

Reply:
See other comments, this is the autogenerated client below our own implementations